controllers/rack.go

```diff
@@ -420,7 +424,14 @@ func (r *SingleClusterReconciler) upgradeOrRollingRestartRack(found *appsv1.Stat
 	}
 
 	if r.aeroCluster.Spec.RackConfig.MaxIgnorablePods != nil {
-		if res := r.handleNSOrDeviceRemovalForIgnorablePods(rackState, ignorablePodNames); !res.isSuccess {
+		if res = r.handleNSOrDeviceRemovalForIgnorablePods(rackState, ignorablePodNames); !res.isSuccess {
+			return found, res
+		}
+	}
+
+	if r.aeroCluster.Spec.K8sNodeBlockList != nil {
+		found, res = r.handleK8sNodeBlockListPods(found, rackState, ignorablePodNames, failedPods)
```

Comment:
Is it possible to trigger this function only if there is any change in `K8sNodeBlockList` to reduce unnecessary computation?